pkg/operator/discover/discover.go

```diff
@@ -212,10 +274,46 @@ func ListDevices(context *clusterd.Context, namespace, nodeName string) (map[str
 			logger.Warningf("failed to unmarshal %s", deviceJson)
 			continue
 		}
-		devices[node] = d
+		for i := range d {
+			devices = append(devices, d[i])
+		}
 	}
-	logger.Debugf("devices %+v", devices)
+	logger.Debugf("devices in use %+v", devices)
 	return devices, nil
+
+}
+
+func FreeDevices(context *clusterd.Context, nodeName, clusterName string) error {
+	if len(nodeName) == 0 || len(clusterName) == 0 {
+		return nil
+	}
+	namespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+	cmName := fmt.Sprintf(deviceInUseCMName, clusterName, nodeName)
+	// delete configmap
+	err := context.Clientset.CoreV1().ConfigMaps(namespace).Delete(cmName, &metav1.DeleteOptions{})
+	if err != nil && !kserrors.IsNotFound(err) {
+		return fmt.Errorf("failed to delete configmap %s/%s. %+v", namespace, cmName, err)
+	}
+
+	return nil
+}
+
+func FreeDevicesByCluster(context *clusterd.Context, clusterName string) error {
```

Comment:
instead of this delete, actually all we need to do is set the `ownerRef` on the configmaps when they are created. We have this pattern in place for all the resources created by the operator. Search for the `SetOwnerRef` method